taskfile/ast/precondition.go

```diff
@@ -2,16 +2,56 @@ package ast
 
 import (
 	"fmt"
-
-	"gopkg.in/yaml.v3"
+	"sync"
 
 	"github.com/go-task/task/v3/errors"
+	"github.com/go-task/task/v3/internal/deepcopy"
+
+	"gopkg.in/yaml.v3"
 )
 
 // Precondition represents a precondition necessary for a task to run
-type Precondition struct {
-	Sh  string
-	Msg string
+type (
+	Preconditions struct {
+		Preconditions []*Precondition
+		mutex         sync.RWMutex
+	}
+
+	Precondition struct {
+		Sh  string
+		Msg string
+	}
```

Comment:
I've moved `Vars` and `Var` into separate files in another PR. Would be good to follow this pattern here.

Reply:
Yeah
I've splitted in two files